activerecord/lib/active_record/nested_attributes.rb

```diff
@@ -245,18 +245,26 @@ class TooManyRecords < ActiveRecordError
     #
     # === Validating the presence of a parent model
     #
-    # If you want to validate that a child record is associated with a parent
-    # record, you can use the +validates_presence_of+ method and the +:inverse_of+
-    # key as this example illustrates:
-    #
-    #   class Member < ActiveRecord::Base
-    #     has_many :posts, inverse_of: :member
-    #     accepts_nested_attributes_for :posts
+    # The +belongs_to+ association by default validates the presence of the parent
+    # record. In this case, do not explicitly attempt to validate the 
+    # presence of the parent model, or its id attribute. An error will result
+    # if the child is being created via the parent's nested attributes..
+    #
+    # If you specify the +:optional+ option on the +belongs_to+ association,
+    # you may also choose to validate the presence of the parent model. This
+    # combination of +:optional+ and explicit validation may be useful for 
+    # conditional validations, for example. Use the +validates+ method with 
+    # <tt>presence: true</tt> and the +:inverse_of+ key as this example 
+    # illustrates:
```

Comment:
In my testing of creating a child model via nested attributes, the extraneous `validates_presence_of` does not cause a problem.  Though it does add an extra error message if the child is actually missing its parent.  Perhaps I am misunderstanding what you mean here?

If we can leave out the "do not explicitly attempt ... an error will result" part though, what about:

```suggestion
    # The +belongs_to+ association validates the presence of the parent model
    # by default. You can disable this behavior by specifying <code>optional: true</code>. 
    # This can be used, for example, when conditionally validating the presence
    # of the parent model:
```

Reply:
You're right, I just tested it and it doesn't throw an error for the basic case such as what I used in that example. I originally came across "foo must exist" errors in code that was using multiple levels of nesting and a combination of single new child models and arrays of child models. I'm not sure what the simplest test case that will cause the error is, but I think your suggestion is good for covering the common case. Committed. Thanks!